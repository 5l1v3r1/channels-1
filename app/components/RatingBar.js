// @flow
import React from 'react';
import { IconThumbUp, IconThumbDown } from './core/Icons';
import { commaSeparateNumber } from '../utils/utils';
import styles from './RatingBar.scss';

const RatingBar = ({
  likeCount,
  dislikeCount,
  like,
  dislike,
  postLike,
  postDislike
}: {
  likeCount: string,
  dislikeCount: string,
  like: boolean,
  dislike: boolean,
  postLike: (type: string) => void,
  postDislike: (type: string) => void
}) => (
  <div className={styles.ratingBar}>
    <p className={styles.statistics}>
      {commaSeparateNumber(likeCount)} <span>likes &</span> {commaSeparateNumber(dislikeCount)} <span>dislikes</span>
    </p>
    <div className={styles.rating}>
      <label>
        <IconThumbUp className={like ? styles.active : styles.inactive} />
        <input type="checkbox" name="like" checked={like} onChange={postLike} />
      </label>
      <label>
        <IconThumbDown className={dislike ? styles.active : styles.inactive} />
        <input type="checkbox" name="dislike" checked={dislike} onChange={postDislike} />
      </label>
    </div>
  </div>
);

export default RatingBar;
