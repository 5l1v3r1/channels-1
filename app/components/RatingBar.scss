@import '../app.global.scss';

.ratingBar {
  display: flex;
  align-items: center;
  justify-content: space-between;
  min-height: 60px;
  padding: 0 1rem;
}

.statistics {
  margin: 0;
  color: $white;

  @include font-size(16);

  span {
    color: $gray;
  }
}

.rating {
  height: 21px;

  > label {
    display: inline-block;
    height: 100%;

    input {
      display: none;
    }
  }

  label:first-child {
    margin-right: 1rem;
  }
}

.active {
  height: 100%;
  fill: $white;
}

.inactive {
  height: 100%;
  fill: $gray;
}
