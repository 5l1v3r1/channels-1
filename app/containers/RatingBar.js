import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import * as activeVideoActions from '../actions/activeVideo';
import * as ratingsActions from '../actions/ratings';
import RatingBar from '../components/RatingBar';
import { getActiveVideoRating } from '../reducers/index';

function mapStateToProps(state, ownProps) {
  return {
    rating: getActiveVideoRating(state),
    activeVideoId: ownProps.activeVideoId,
    likeCount: ownProps.likeCount,
    dislikeCount: ownProps.dislikeCount
  };
}

const actions = {
  ...activeVideoActions,
  ...ratingsActions
};

function mapDispatchToProps(dispatch) {
  return bindActionCreators(actions, dispatch);
}

export default connect(mapStateToProps, mapDispatchToProps)(RatingBar);
