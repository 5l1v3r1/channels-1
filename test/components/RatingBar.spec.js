import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import RatingBar from '../../app/components/RatingBar';

function setup() {
  const props = {
    likeCount: '',
    dislikeCount: '',
    like: false,
    dislike: false,
    postLike: jest.fn(),
    postDislike: jest.fn()
  };

  const enzymeWrapper = shallow(<RatingBar {...props} />);

  return {
    props,
    enzymeWrapper
  };
}

describe('RatingBar', () => {
  it('should render self', () => {
    const { enzymeWrapper } = setup();
    const tree = toJson(enzymeWrapper);

    expect(tree).toMatchSnapshot();
  });

  it('should call postLike', () => {
    const { props, enzymeWrapper } = setup();
    const input = enzymeWrapper.find('input[name="like"]');

    input.simulate('change');

    expect(props.postLike).toBeCalled();
  });

  it('should call postDislike', () => {
    const { props, enzymeWrapper } = setup();
    const input = enzymeWrapper.find('input[name="dislike"]');

    input.simulate('change');

    expect(props.postDislike).toBeCalled();
  });

  it('like input should be checked', () => {
    const { enzymeWrapper } = setup();
    enzymeWrapper.setProps({
      like: true
    });
    const inputLike = enzymeWrapper.find('input[name="like"]');
    const inputDislike = enzymeWrapper.find('input[name="dislike"]');

    expect(inputLike.props().checked).toBe(true);
    expect(inputDislike.props().checked).toBe(false);
  });

  it('dislike input should be checked', () => {
    const { enzymeWrapper } = setup();
    enzymeWrapper.setProps({
      dislike: true
    });
    const inputLike = enzymeWrapper.find('input[name="like"]');
    const inputDislike = enzymeWrapper.find('input[name="dislike"]');

    expect(inputLike.props().checked).toBe(false);
    expect(inputDislike.props().checked).toBe(true);
  });
});
