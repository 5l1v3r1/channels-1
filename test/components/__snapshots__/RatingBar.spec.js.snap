// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RatingBar should render self 1`] = `
<div
  className="ratingBar"
>
  <p
    className="statistics"
  >
    0
     
    <span>
      likes &
    </span>
     
    0
     
    <span>
      dislikes
    </span>
  </p>
  <div
    className="rating"
  >
    <label>
      <Component
        className="inactive"
      />
      <input
        checked={false}
        name="like"
        onChange={[Function]}
        type="checkbox"
      />
    </label>
    <label>
      <Component
        className="inactive"
      />
      <input
        checked={false}
        name="dislike"
        onChange={[Function]}
        type="checkbox"
      />
    </label>
  </div>
</div>
`;
